app/tray/Account/Requests/TransactionRequest/TxMainNew/overview.js

```diff
@@ -17,16 +18,23 @@ const SimpleContractCallOverview = ({ method }) => {
 }
 
 const ApproveOverview = ({ amount, decimals, symbol }) => {
+  const isRevoke = BigNumber(amount).isZero()
   return (
     <div>
-      <span>{'Approve Spending '}</span>
-      <DisplayValue
-        type='ether'
-        value={amount}
-        valueDataParams={{ decimals }}
-        currencySymbol={symbol}
-        currencySymbolPosition='last'
-      />
+      {isRevoke ? (
+        <span>{`Revoke Approval for ${symbol} `}</span>
```

Comment:
do we need the space at the end here?

Reply:
in general I think any functionality-specific components (like this) should stay in the `app` section and then we would have more general use components (like `CustomInputField`) in the `resources/Components` section, but we dont need to make all those changes right now. might just be worth a rename as a good compromise